bignum.cc

```diff
@@ -431,16 +489,63 @@ NAN_METHOD(BigNum::ToString)
     base = tbase;
   }
   char *to = NULL;
-  switch (base) {
-  case 10:
-    to = BN_bn2dec(&bignum->bignum_);
-    break;
-  case 16:
-    to = BN_bn2hex(&bignum->bignum_);
-    break;
-  default:
-    Nan::ThrowError("Invalid base, only 10 and 16 are supported");
-    return;
+  if (base == 2) {
```

Comment:
It seems a little wasteful to special-case when you can just generalize `BN_bn2base36(a)` to `BN_bn2base(a, base)` where `2 <= base <= 36`.